src/buffer_manager.ts

```diff
@@ -133,7 +133,14 @@ export class BufferManager implements Disposable {
      */
     private bufferProvider: BufferProvider;
 
+    /**
+     * Used to debounce the editor options change event
+     */
     private editorOptionsChangedTimers: WeakMap<TextEditor, NodeJS.Timeout> = new WeakMap();
+    /**
+     * Avoids unnecessary sending of options to neovim
+     */
+    private editorOptionsMap = new WeakMap<TextEditor, TextEditorOptions>();
```

Comment:
can we check some other existing state (i.e. whatever these settings are actually used for), instead of adding this cache?

Reply:
It seems not. The events provided by VSCode only include the new value after the change.